bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/handler/MideaACHandler.java

```diff
@@ -0,0 +1,613 @@
+            } else if (channelUID.getId().equals(CHANNEL_TEMPERATURE_UNIT)) {
+                connectionManager.sendCommand(CommandHelper.handleTempUnit(command, lastresponse), callbackLambda);
+            } else if (channelUID.getId().equals(CHANNEL_SLEEP_FUNCTION)) {
+                connectionManager.sendCommand(CommandHelper.handleSleepFunction(command, lastresponse), callbackLambda);
+            } else if (channelUID.getId().equals(CHANNEL_ON_TIMER)) {
+                connectionManager.sendCommand(CommandHelper.handleOnTimer(command, lastresponse), callbackLambda);
+            } else if (channelUID.getId().equals(CHANNEL_OFF_TIMER)) {
+                connectionManager.sendCommand(CommandHelper.handleOffTimer(command, lastresponse), callbackLambda);
+            } else if (channelUID.getId().equals(CHANNEL_MAXIMUM_HUMIDITY)) {
+                connectionManager.sendCommand(CommandHelper.handleMaximumHumidity(command, lastresponse),
+                        callbackLambda);
+            }
+        } catch (MideaConnectionException | MideaAuthenticationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (MideaException | IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    /**
+     * To initialize an AC Thing, first check if discovery on your LAN has
+     * been completed. In the discovery process the IP address, IP port, device ID
+     * and version are established. Next for V.3 devices, if the token and key are not
+     * known, the cloud needs to be contacted to get the token and key using either
+     * the default NetHome Plus cloud or your own cloud account with your password and
+     * email. LAN discovery DOES NOT include the token key retrieval needed to communicate
+     * with the AC. V2 devices bypass the cloud connection step because they use a simplier
+     * hard-coded encryption. Next the Connection Manager is established. Then a command
+     * is formed and sent to retrieve the AC capabilities if they have not been
+     * discovered. Capabilities are not returned in the initial LAN Discovery.
+     * Lastly the routine polling, token key update and Energy polling frequency are set.
+     * 
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(MideaACConfiguration.class);
+
+        // Check for valid discovery configurations and discover again if not
+        if (!config.isValid()) {
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_PENDING, "Configuration not valid");
+            if (config.isDiscoveryPossible()) {
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_PENDING,
+                        "Configuration missing, discovery needed. Discovering...");
```

Comment:
Please use i18n custom keys when setting the thing state. This would make it possible to add translations.
For reference: https://next.openhab.org/docs/developer/utils/i18n.html#using-custom-keys

(applies to all occurences of `updateStatus` with a thingstatusdetail)

Reply:
Just to check. I have run `mvn i18n:generate-default-translations -pl :org.openhab.binding.mideaac`, but need to manually add the thing status details in the MideaACHandler.

Reply:
I think I have this sorted out. With the limited slate of states and descriptions did some consolidation and removed some unnecessary ones. I have assumed that `e.getMessage()` override the content in the i18n.